packages/kotti-ui/source/kotti-field-inline-edit/KtFieldInlineEdit.vue

```diff
@@ -0,0 +1,339 @@
+			() => props.placeholder ?? translations.value.placeholder,
+		)
+
+		const preventConfirm = computed<boolean>(() => {
+			const fieldValidation = field.validation.type
+
+			switch (props.preventConfirmationOn) {
+				case KottiFieldInlineEdit.ConfirmationValidation.ERROR:
+					return fieldValidation === 'error'
+				case KottiFieldInlineEdit.ConfirmationValidation.WARNING:
+					return ['error', 'warning'].includes(fieldValidation)
+				case KottiFieldInlineEdit.ConfirmationValidation.NEVER:
+					return false
+				default:
+					throw new Error(
+						`KtFieldInlineEdit: Unexpected type for 'preventConfirmationOn': ${props.preventConfirmationOn}`,
+					)
+			}
+		})
+
+		const setFieldValue = (newValue: KottiFieldInlineEdit.Value) => {
+			field.setValue(newValue)
+			forceUpdate()
+		}
+
+		const updateIsEditing = (shouldEdit: boolean) => {
+			if (shouldEdit === props.isEditing) {
+				/*
+					condition shouldn't be moved outside of the wrapping if.
+					In case of not emitting the update:isEditing, the component
+					has to re-focus on the input field if needed.
+
+					Otherwise, the component emits and if the isEditing is set by
+					the user app, the same condition will be applied in the watcher.
+				*/
+				if (shouldEdit && !isInFocus(inputRef.value)) inputRef.value?.focus()
+
+				return
+			}
+
+			emit('update:isEditing', shouldEdit)
+		}
+
+		watch(
+			() => props.isEditing,
+			(shouldEdit, wasEditing) => {
+				if (shouldEdit && !isInFocus(inputRef.value)) inputRef.value?.focus()
+
+				if (shouldEdit === wasEditing) return
+
+				preEditingValue.value = props.value
+			},
+			{ immediate: true },
+		)
+
+		const onConfirm = () => {
+			if (preventConfirm.value) {
+				if (!isInFocus(inputRef.value)) inputRef.value?.focus()
+				return
+			}
+
+			emit('confirm', field.currentValue)
+			updateIsEditing(false)
+
+			blurField(fieldRef.value)
+		}
+
+		const onCancel = () => {
+			setFieldValue(preEditingValue.value)
+			updateIsEditing(false)
+
+			blurField(fieldRef.value)
+		}
+
+		const onClick = (event: MouseEvent | KeyboardEvent) => {
+			if (event.target === null || props.isDisabled) return
+
+			const isClickOutside = !isFieldTriggered(fieldRef.value, event.target)
+
+			if (isClickOutside) onCancel()
+			else updateIsEditing(true)
+		}
+
+		const lastFocusTarget = ref<EventTarget | null>(null)
+		/**
+		 * `tab` key presses don't trigger on mobile,
+		 * and @blur events are tricky to implement given the implementation of
+		 * `KtField`. Therefore, a listener is attached to DOM `focus` events.
+		 *
+		 * The target element is continuously maintained in a variable to compare
+		 * changes and distinguish between ignore and confirm cases.
+		 *
+		 * The focus event is captured before the click in case of clicking a focusable element.
+		 */
+		const onFocusChange = (event: Event) => {
+			if (event.target === null || props.isDisabled) return
+
+			event.stopPropagation()
+			event.preventDefault()
```

Comment:
i think this should be removed too
i tested on the comments docs and the focus on other components on the same page is also broken and doesn't trigger